### PR TITLE
Update Catapult URL to v2 if not already

### PR DESCRIPTION
**Pull request by jakegut**
https://clever.atlassian.net/browse/INFRANG-4774

Tested the scripts on https://github.com/Clever/catalog-sync-service/tree/jake-circleci-test

### CODE REVIEW

File: circleci/catapult-publish-lambda [hunk 1]
```diff
@@ -51,9 +51,11 @@ done
 
 CIRCLE_CI_INTEGRATIONS_URL=$(dirname $CATAPULT_URL)
 
-# hack to switch from /catapult to /v2/catapult
 # TODO: should probably `ark init` things with an environment variable pointing to base url of circle-ci-integrations
-CATAPULT_URL=$(echo "${CATAPULT_URL}" | sed 's/\/catapult/\/v2\/catapult/')
+# Update Catapult URL to v2 if not already
+if [[ ! $CATAPULT_URL = *"/v2/catapult"* ]]; then 
+  CATAPULT_URL=${CATAPULT_URL/catapult/v2/catapult}
```
Comment (taylor-sutton) [inline circleci/catapult-publish-lambda:57]: I'm a little confused why this worked - the URL is `${BASE}/v2/catapult` (https://github.com/Clever/circle-ci-integrations/blob/0028020205ec365e4eec2e36eee5813b1cc6a679/swagger.yml#L50) not `/catapult/v2/catapult`)

For example from the `sed` on the left-hand-side:
```
at 14:07:49 ❯ echo 'base/catapult' | sed 's/\/catapult/\/v2\/catapult/'
base/v2/catapult
```

Comment (jakegut) [reply circleci/catapult-publish-lambda:57]: This is a find-replace feature in bash, it's not appending in this case. More about it here: https://stackoverflow.com/a/13210909

Comment (taylor-sutton) [reply circleci/catapult-publish-lambda:57]: Ooo the `/` is *inside the brackets*, thanks for explaining.

**Review by taylor-sutton — APPROVED** [review]
LGTM! I wasn't sure if you did this already, but as extra validation I compared the builds stored in catapult for `catalog-sync-service` master branch vs your test branch using `curl --location --request GET 'https://production--catapult.int.clever.com/v2/applications/catalog-sync-service/builds/aed70d4'` and curl --location --request GET 'https://production--catapult.int.clever.com/v2/applications/catalog-sync-service/builds/a53ffa8' and the differences were only the expected ones (branch, creation date, ID, etc)

**Comment by jakegut** [comment]
> I compared the builds stored in catapult

Thanks for doing this, I didn't really know you could do this at all.

**Pull request merged**